sync/class.jetpack-sync-actions.php

```diff
@@ -439,18 +439,41 @@ static function cleanup_on_upgrade( $new_version = null, $old_version = null ) {
 		}
 	}
 
-	static function get_sync_status() {
+	static function get_sync_status( $fields = null ) {
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-wp-replicastore.php';
 		self::initialize_sender();
 
 		$sync_module     = Jetpack_Sync_Modules::get_module( 'full-sync' );
 		$queue           = self::$sender->get_sync_queue();
 		$full_queue      = self::$sender->get_full_sync_queue();
 		$cron_timestamps = array_keys( _get_cron_array() );
 		$next_cron       = $cron_timestamps[0] - time();
+		$store           = new Jetpack_Sync_WP_Replicastore();
+
+		$checksums = array();
+
+		if ( is_null( $fields ) === false ) {
+			$fields_params = explode( ',', $fields );
```

Comment:
I'd suggest trimming the values as well, it's a common mistake to pass a `fields` argument with spaces after the commas, for example `?fields=posts_checksum, comments_checksum`. I know that's incorrect syntax, but it's cheap to guard against it.

```suggestion
			$fields_params = array_map( 'trim', explode( ',', $fields ) );
```

Reply:
Excellent idea. Thanks.